Core/Dlr/Actions/DynamicMetaObjectBinder.cs

```diff
@@ -165,6 +167,7 @@ public abstract class DynamicMetaObjectBinder : CallSiteBinder {
             return mos;
         }
 
+#if FEATURE_REMOTING
         private static BindingRestrictions AddRemoteObjectRestrictions(BindingRestrictions restrictions, object[] args, ReadOnlyCollection<ParameterExpression> parameters) {
 #if !SILVERLIGHT
```

Comment:
We should emulate !Silverlight here. It's the same semantics.
```suggestion
        private static BindingRestrictions AddRemoteObjectRestrictions(BindingRestrictions restrictions, object[] args, ReadOnlyCollection<ParameterExpression> parameters) {
#if !SILVERLIGHT && FEATURE_REMOTING
```

Reply:
But isn't the whole method call useless without remoting (yes - same goes for the existing code I guess)?

Reply:
Yes, it is. Thing is, same goes for the Silverlight variant from when Microsoft first created this code. So, there's the symetry aspect to it. 

As for a deeper reasoning when you would do it without precendent in the code, hmm...you could argue that for keeping the method call even it it's a NOP to keep the conditional code as straight forward as possible. If you drop an entire method instead of turning it into a NOP, you have to make sure the call site continues to behave consistently. Fortuantely, we don't have an IF here at the call site, but if we did, we'd then have to think if the IF should be made conditional, too, and so on. This would increase complexity or require different approaches for each place that gets a preprocessor condition. 

But mostly, it's about just sticking to the already existing pathways instead of forging new ones unless that's actually necessary.